station-items-v0/src/main/java/net/modificationstation/stationapi/api/item/StationItemStack.java

```diff
@@ -1,13 +1,20 @@
 package net.modificationstation.stationapi.api.item;
 
+import net.minecraft.entity.Living;
+import net.minecraft.entity.player.PlayerBase;
 import net.minecraft.util.io.CompoundTag;
 import net.modificationstation.stationapi.api.item.nbt.StationNBT;
 import net.modificationstation.stationapi.api.util.Util;
+import net.modificationstation.stationapi.mixin.item.MixinPlayerBase;
 
 public interface StationItemStack extends StationNBT {
 
     @Override
     default CompoundTag getStationNBT() {
         return Util.assertImpl();
     }
+
+    default boolean preHit(Living otherEntity, PlayerBase player){ return Util.assertImpl(); }
+
+    default boolean preMine(int x, int y, int z, int l, Living entity){ return Util.assertImpl(); }
```

Comment:
1. No block ID?
2. `l` is the side of the block that is hit, so probably should name it that. Though it's not present in vanilla `postMine` in the first place.
3. Vanilla `postMine` takes a `PlayerBase`, not `Living`.

Reply:
1. Added a BlockState to the method as a parameter
2. Renamed the parameter to side (wasn't sure about if this is a side, thanks for clarifying)
3. Changed all the parameters that are always a player to PlayerBase